src/cunumeric/sort/sort.cu

```diff
@@ -19,6 +19,7 @@
 #include "cunumeric/sort/cub_sort.h"
 #include "cunumeric/sort/thrust_sort.h"
 #include "cunumeric/utilities/thrust_allocator.h"
+#include "cunumeric/utilities/thrust_util.h"
 
 #include <thrust/scan.h>
```

Comment:
@mfoerste4 can you take a look at this file and confirm that thrust not implicitly synchronizing its calls would be ok? IIUC there is an issue with destroying `DeferredBuffer`'s eagerly, as those deletions are not by default synchronized with the task's stream.

Also note that `thrust::reduce` will always synchronize, even under `par_nosync` mode.